### ISSUE TITLE
argument to `.next()` and more

### ISSUE THREAD

**Issue by elibarzilay** [mozilla/sweet.js#534]
(Sorry about the lump of bugs & requests, but they're all related.)

1. The string argument to `.next()` is not described somewhere.  I had to use the source.

2. Why is there no `.next("Identifier")`?  -- IIUC, that would conveniently be similar to doing some verbose dance of

    let id = ...;
    if (!id.isIdentifier())
      throw Error(`maymacro: expected identifier, got ${?something?}`)

3. Related to that, I didn't find anything that could be written at the end of that.

4. Related to that, it would be nice to have some variant of Racket's `raise-syntax-error`, with a similar api.  That is, something that always adds a "foo:" prefix for the name of the currently-transforming macro name, throws the error with source location and whatever (when available), and also if there is no nice way to display these things (see last item) then such a utility can do something about that too.

**Comment by disnet** [comment]
Indeed, `.next` should accept all the JS grammar productions but currently only knows about `AssignmentExpressions` (with alias `expr` since that's the one you usually want), `Expression`, and `Syntax`. The enforester needs a little love before adding these to `.next` since I think there are a few `enforest*` methods that don't quite align with the actual grammar rules.

And yeah something like `raise-syntax-error` would be great. I've been holding off on adding stuff like that before I get modules landed because otherwise we'd have to hack them in as macro "primitives".

**Comment by elibarzilay** [comment]
Re the error thing: I think that it's a good idea to wait with it (same as with some rule-based macro generators, as I discussed in another issue).  But it does need some built-in-ness kind of magic to at least know the name of the currently-expanding macro.  (So, either that, or make that name available to user macros.)
